internal/runners/push/push.go

```diff
@@ -170,61 +174,93 @@ func (r *Push) Run(params PushParams) (rerr error) {
 		}
 
 		r.out.Notice(locale.Tl("push_creating_project", "Creating project [NOTICE]{{.V1}}[/RESET] under [NOTICE]{{.V0}}[/RESET] on the ActiveState Platform", targetNamespace.Owner, targetNamespace.Project))
-		targetPjm, err = model.CreateEmptyProject(targetNamespace.Owner, targetNamespace.Project, r.project.Private())
+
+		// Create a new project with the current project's buildexpression.
+		expr, err := bp.GetBuildExpression(r.project.Owner(), r.project.Name(), commitID.String())
 		if err != nil {
-			return errs.Wrap(err, "Failed to create project %s", r.project.Namespace().String())
+			return errs.Wrap(err, "Could not get buildexpression")
+		}
+		newCommitID, err = bp.CreateProject(&model.CreateProjectParams{
+			Owner:       targetNamespace.Owner,
+			Project:     targetNamespace.Project,
+			Private:     r.project.Private(),
+			Description: locale.T("commit_message_add_initial"),
+			Expr:        expr,
```

Comment:
The new project does not have any history. It has a single entry that has the current state of the project being cloned. We cannot attach a project's existing commit to a new, empty project because the branches don't match (you get a "The branch you're trying to update has changed remotely, please run `state pull`).

Does this answer your question?

Reply:
What I'm wondering is if we run this flow with a previous version of the state tool will the newly created project have more history than just the local commit we pushed? If that's the case then this update will change that behaviour.

Reply:
@MDrakos Yes, it looks like pushes via the VCS preserves history, while attaching commits via buildplanner loses individual history (the end-result is still the same).

Do we need to create an empty project, walk through history, and attach staged commits individually to preserve history?

Reply:
IMO no, but that is a question for @rawktron. This is a _very_ niche flow that I doubt is being used _and_ we have `state fork` for creating a new project and preserving the history. But let's double-check with product before we move forward.

Reply:
I don't think we need to do that much to preserve history. I think we discussed this previously and were ok with the fact that your commits are effectively squashed when you do this. And that is ok. @mitchell-as